### PR TITLE
fix: handle invalid and small prices

### PR DESCRIPTION
**Pull request by OnlyJousting**
Changes:
- Filter invalid and negative candle prices before calculating chart ranges
- Keep chart padding from pushing non-negative price ranges below zero
- Improve small price tick formatting for sub-unit assets
- Simplify chart price formatting by using locale rounding directly
- Add focused tests for range calculation and price formatting edge cases

### CODE REVIEW

**Review by 0xtutti — APPROVED** [review]

File: packages/curve-ui-kit/src/features/candle-chart/utils.ts [hunk 1]
```diff
@@ -106,24 +109,16 @@ export const priceFormatter = (x: number, delta: number) => {
...
+  // Let a tight visible range drive precision, while allowing extra decimals for sub-unit assets.
   const magnitude = deltaMagnitude < valueMagnitude - 1 ? deltaMagnitude : valueMagnitude
-  const decimals = Math.min(4, Math.max(0, 3 - magnitude))
+  const maxDecimals = x < 1 ? MAX_SUB_UNIT_PRICE_DECIMALS : MAX_PRICE_DECIMALS
+  const decimals = Math.min(maxDecimals, Math.max(0, 3 - magnitude))
 
-  const formatted = formatNumber(x, {
-    decimals,
-    abbreviate: false,
+  return x.toLocaleString('en-US', {
```
Comment (0xAlunara) [inline packages/curve-ui-kit/src/features/candle-chart/utils.ts:118]: rip to using `formatNumber`

**Review by 0xAlunara — DISMISSED** [review]
Presentation code is always hairy, so if it works it works despite code quality 🫠

**Review by 0xtutti — DISMISSED** [review]

File: packages/curve-ui-kit/src/features/candle-chart/utils.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
-import { TIME_OPTION_MS } from '@ui-kit/lib/model/time'
-import { formatNumber } from '@ui-kit/utils'
+import { TIME_OPTION_MS } from '../../lib/model/time'
+import { formatNumber } from '../../utils/number'
 import type { TimeOption } from './types'
 
 const toSeconds = (timeOption: TimeOption) => TIME_OPTION_MS[timeOption] / 1000
+const isValidPrice = (value: number) => Number.isFinite(value) && value >= 0
```
Comment (DanielSchiavini) [inline packages/curve-ui-kit/src/features/candle-chart/utils.ts]: why do we expect invalid values? A comment would be nice

File: packages/curve-ui-kit/src/features/candle-chart/utils.ts (hunk 1 above)
Comment (OnlyJousting) [reply packages/curve-ui-kit/src/features/candle-chart/utils.ts:118]: the reason is to allow rounding to 0 in the chart instead of "<0.00001" etc, 0 is more desirable for chart ticks

File: packages/curve-ui-kit/src/features/candle-chart/utils.ts (hunk 2 above)
Comment (OnlyJousting) [reply packages/curve-ui-kit/src/features/candle-chart/utils.ts]: hm yea this might be covering up issues in the data which I guess is counter to what we would want, we want to see those issues. I'll remove it (I don't expect invalid values)

**Review by 0xAlunara — APPROVED** [review]

**Review by 0xtutti — APPROVED** [review]

**Review by DanielSchiavini — APPROVED** [review]
